ledger/eval/eval.go

```diff
@@ -1179,15 +1179,9 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 	txid := txn.ID()
 
 	if eval.validate {
-		err = eval.block.Alive(txn.Txn.Header)
-		if err != nil {
-			return err
-		}
-
-		// Transaction already in the ledger?
-		err = cow.checkDup(txn.Txn.FirstValid, txn.Txn.LastValid, txid, ledgercore.Txlease{Sender: txn.Txn.Sender, Lease: txn.Txn.Lease})
-		if err != nil {
-			return err
+		// run Alive, WellFormed, and checkDup checks
```

Comment:
we call `eval.transaction` with `cow := eval.state.child(len(txgroup))` so it is `eval.state.child.checkDup` and `TestTransaction` runs `eval.state.checkDup`.